src/ert/ensemble_evaluator/builder/_legacy.py

```diff
@@ -219,7 +225,7 @@ async def _evaluate_inner(
                     dispatch_url=self._config.dispatch_uri,
                     cert=self._config.cert,
                     token=self._config.token,
-                    experiment_id=self.id_,
+                    experiment_id=experiment_id,
```

Comment:
Shall we then require that also `experiment_id is not None`?:
```python
if output_bus and experiment_id:  # when running experiment server
        self._job_queue.add_dispatch_information_to_jobs_file(
            ens_id=self.id_,
            dispatch_url=self._config.dispatch_uri,
            cert=self._config.cert,
            token=self._config.token,
            experiment_id=experiment_id,
        )
```

Reply:
Yepp, can do! and Done!